### ISSUE TITLE
gnat fails to build which in turn breaks ghdl

### ISSUE THREAD

**Issue by ivantomac** [NixOS/nixpkgs#173830]
### Steps To Reproduce
1. `nix-build '<nixpkgs>' -A ghdl`

### Build log
This the end of the log, once configure finishes and tries to create Makefiles. I didn't notice any errors or anything notable in the configuration section.

```
configure: creating ./config.status
yes
config.status: creating Makefile
checking for working strncmp... config.status: creating testsuite/Makefile
config.status: creating config.h
config.status: executing default commands
yes
configure: updating cache ./config.cache
configure: creating ./config.status
config.status: creating Makefile
config.status: creating testsuite/Makefile
config.status: creating config.h
config.status: executing default commands
make[2]: Leaving directory '/build/build'
make[1]: *** [Makefile:21252: stage1-bubble] Error 2
make[1]: Leaving directory '/build/build'
make: *** [Makefile:21573: bootstrap] Error 2
error: builder for '/nix/store/1rya10nw23lm6im8frkk5z3ic9ibb80m-gnat-6.5.0.drv' failed with exit code 2;
       last 10 log lines:
       > configure: updating cache ./config.cache
       > configure: creating ./config.status
       > config.status: creating Makefile
       > config.status: creating testsuite/Makefile
       > config.status: creating config.h
       > config.status: executing default commands
       > make[2]: Leaving directory '/build/build'
       > make[1]: *** [Makefile:21252: stage1-bubble] Error 2
       > make[1]: Leaving directory '/build/build'
       > make: *** [Makefile:21573: bootstrap] Error 2
       For full logs, run 'nix log /nix/store/1rya10nw23lm6im8frkk5z3ic9ibb80m-gnat-6.5.0.drv'.
error: 1 dependencies of derivation '/nix/store/97fcrz1574q78y8y0pnafijdzmhqdf83-gnat-wrapper-6.5.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/g7q32ml0pnvrk27rm3cyny467qzl79s5-gnat-11.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/skfpizi2cmg28v3bm083hikc1ad5s10y-gnat-wrapper-11.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/k06165bcqz6zapkx9ygpj8rg1hkqhjnv-ghdl-mcode-1.0.0.drv' failed to build
```

### Notify maintainers
@Lucus16 @thoughtpolice 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.7, NixOS, 22.05 (Quokka), 22.05pre378655.48037fd9042`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(user): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by vcunat** [comment]
Upstream report: https://sourceware.org/bugzilla/show_bug.cgi?id=29162

**Comment by collares** [comment]
Will be fixed by #173893.
